### ISSUE TITLE
RemoveSwitchMutator: false positive when case points to default

### ISSUE THREAD

**Issue by rnveach** [hcoles/pitest#1137]
Project: https://github.com/checkstyle/checkstyle

Mutations: All at https://github.com/checkstyle/checkstyle/blob/master/config/pitest-suppressions/pitest-annotation-suppressions.xml

Command: mvn test-compile org.pitest:pitest-maven:mutationCoverage -Ppitest-annotation

Source 1:
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck.java#L442

The report says case 1 through 3 is killed, but 4 is survived. This is an enum and is going  against the source code ordering of the enums, so it references `IGNORE` which is attached to the `default` case. All the other enums are verified as being resolved. Moving `IGNORE` to any other case label results in tests failures. There is no way to kill this mutation since the case is directly attached to the default case.

Source 2:
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/SuppressWarningsCheck.java#L250-L251

`TokenTypes.IDENT` and `TokenTypes.DOT` are the ones with the specific case values. While they are not directly attached to the default, they do have the same code as the default. I tried changing the code so all 3 cases are attached, but it still survives and thus there is no way to kill the mutation.

I would expect these mutations to not even be generated since they are basically non-killable no matter what is done during execution. It would be better if pitest was smart and saw the code between the movable case and the default is the same code, but I am ok if we just ignore cases directly tied to the default.

**Comment by hcoles** [comment]
Having taken a quick look at this, I think it should be possible to filter out equivalent mutants of the form

```java
      switch (value) {
        case 1:
          // do stuff
         break;
        case 2:
          // do things
         break;
        case 3:
          // fall through
        default:
         // do something else
      }
```

Where the third case is redundant, but added by the programmer as a matter of style.

For the simple examples I've looked at, the compiler adds a jump to the same location as in the default, so this should be easy to filter (so long as the compiler does this consistently which has yet to be confirmed).

This will only work for fall throughs. Adding the redundant break in the code below would result in jumps to different locations and prevent the filtering from happening.

```java
      switch (value) {
        case 1:
          // do stuff
         break;
        case 2:
          // do things
         break;
        case 3:
          break; // redundant
        default:
         // do nothing
      }
```

**Comment by rnveach** [comment]
> This will only work for fall throughs. Adding the redundant break in the code below would result in jumps to different locations and prevent the filtering from happening.

Yes, this is fine. This is why my other point was if the code in the cases were identical which I would assume is more complex to check and I am fine with avoiding this for now to just handle the very simple case. We can merge identical code cases to get around it.

**Comment by rnveach** [comment]
@hcoles Thanks for releasing the newest pitest. We are doing an update at https://github.com/checkstyle/checkstyle/pull/12707 and it is resolving mutations that we expected.

The following is not be resolved.
````
        case 3:
          break; // redundant
        default:
         // do nothing
````

We can change the code for this if pitest will not handle such a case.

I am not seeing anything more and the issue can be closed if you are good on your end too.
